lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -253,16 +253,16 @@ static void buildModule(OpBuilder &builder, OperationState &result,
 
   // Record the names of the arguments if present.
   SmallString<8> attrNameBuf;
-  for (size_t i = 0, e = ports.size(); i != e; ++i) {
+  // Record the names of the arguments if present.
+  SmallVector<Attribute> argNames;
+  for (size_t i = 0, e = ports.size(); i != e; ++i)
     if (ports[i].getName().empty())
-      continue;
-
-    auto argAttr =
-        NamedAttribute(builder.getIdentifier("firrtl.name"), ports[i].name);
+      argNames.push_back(builder.getStringAttr(""));
+    else
+      argNames.push_back(ports[i].name);
```

Comment:
I don't think you need to check for an empty name anymore.. 
```
  for (size_t i = 0, e = ports.size(); i != e; ++i)
    argNames.push_back(ports[i].name);
```

Reply:
you can resize the argNames array to be the same size as the ports array

Reply:
The port.name might not exist, so I think we would need the empty check, 
```
 struct ModulePortInfo {
... 
   StringRef getName() const { return name ? name.getValue() : ""; } 

```

Reply:
The `getName` method should be changed.  We should enforce the invariant that the portNames list is the same size as the # ports in a firrtl module, and always has (possibly empty?) StringAttr members in its array.  This will simplify a bunch of this code and make it less flakey.